### PR TITLE
[YUNIKORN-2515] Add property event.RESTResponseSize to the batch event handler

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Update the code with the property `event.RESTResponseSize` as per the design document.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2515

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by pbacsko** [comment]
I think `Test_fixedRule_ruleDAO` needs to be corrected, the map iteration order sometimes changes.

**Comment by pbacsko** [comment]
Created #887 to fix the test case. Will rebase this PR as soon as that one gets merged.

**Comment by chia7712** [comment]
I have another idea of implementing this new property.

 `event.RESTResponseSize` can be applied to `EventSystemImpl` and thus `GetEventsFromID` can pick up `min(count, RESTResponseSize)` to generate `EventRecord`. That is similar to `requestCapacity` and `ringBufferCapacity` WDYT?

https://github.com/apache/yunikorn-core/blob/d46f216e2fa85a38edff11d48e82f481a0c03f3c/pkg/events/event_ringbuffer.go#L104

**Comment by pbacsko** [comment]
> `event.RESTResponseSize` can be applied to `EventSystemImpl` and thus `GetEventsFromID` can pick up `min(count, > RESTResponseSize)` to generate `EventRecord`. That is similar to `requestCapacity` and `ringBufferCapacity` WDYT?

Hm... I'd rather keep this separate. This property concerns the REST output, I think it's better to handle it separately from the event system.

File: pkg/webservice/handlers.go [hunk 1]
```diff
@@ -83,6 +85,35 @@ func init() {
 	allowedActiveStatusMsg = fmt.Sprintf("Only following active statuses are allowed: %s", strings.Join(activeStatuses, ","))
 
 	streamingLimiter = NewStreamingLimiter()
+
+	configs.AddConfigMapCallback("rest-response-size", func() {
+		newSize := configs.DefaultRESTResponseSize
+		configMap := configs.GetConfigMap()
+		if value, ok := configMap[configs.CMRESTResponseSize]; ok {
```
Comment (chia7712) [inline pkg/webservice/handlers.go:92]: How about reusing `common.GetConfigurationUint`?

**Review by chia7712 — COMMENTED** [review]
@pbacsko thanks for updated PR. overall LGTM. one last comment :)

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**